### PR TITLE
Added checkstyle plugin to main POM.xml with basic config.

### LINKED ISSUE #397: Add Checkstyle Plugin to Maven Build
For each PR certain quality criteria have to be met  to get merged.
One of these criteria is code formatting (e.g. indentation by spaces, not adding extra whitespace etc.)

### Expected Behavior

One should be able to check if these criteria are met by a report by calling Maven `mvn checkstyle:checkstyle `.
If so, then a report will be available in `target/site/checkstyle.html`.

The rules and conventions shall be configured in project root directory inside `checkstyle.xml`, so that the configuration
can evolve independenly from actual build configuration.

At this time, Checkstyle should not break the build, but this is definitely the way to go when we've found a good set of conventions.

### Current Behavior

There is no style check plugin defined hence there is no way to perform automated code checks.

### Context

I've now posted multiple PRs and always I'm strugelling with whitespaces, empty lines etc.
Also desired code formatting style changes in different areas of the project, so that it is not really easy to anticipate the desired style.

### PR DESCRIPTION
**Pull request by Oliver-Loeffler**
The [Checkstyle](https://checkstyle.org) [Maven Plugin](https://maven.apache.org/plugins/maven-checkstyle-plugin/index.html) has been added to the main `POM.xml` to enable creation of a code style report.

A very basic configuration is provided as well, which primarily checks for extra whitespace, empty lines and empty blocks.
This plugin is not yet executed during build.

### Issue

Fixes #397 

### Progress

- [ ] Change must not contain extraneous whitespace
- [ ] License header year is updated, if required
- [x] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

**Comment by abhinayagarwal** [comment]
We should start with a subset of current proposed rules in `checkstyle.xml`. Rules can be added once current violations are fixed.

For the current set of rules, there are almost 32.5K and 3.3K violations in kit and app respectively.

**Comment by Oliver-Loeffler** [comment]
This is strange, there is something like 35 files for app and 645 for kit.
With that many violations the given configuration works differently on our systems.
Did I configure it incorrectly?

But yes, I agree to start with a small config and grow it over time.

**Comment by Oliver-Loeffler** [comment]
The numbers you mention are created when running with the standard `sub-checks` configuration. 
This was tested using Maven 3.6.3 - so I'll repeat with 3.8.2.

**Comment by Oliver-Loeffler** [comment]
Okay, I pulled issue-397 into a fresh directory. Re-ran everything using Maven 3.8.2 with `mvn checkstyle:checkstyle` and got the attached results. It seems to stick with the `checkstyle.xml` provided in the project and configured in POM.
[CheckstyleResults.zip](https://github.com/gluonhq/scenebuilder/files/7081810/CheckstyleResults.zip)

**Comment by abhinayagarwal** [comment]
I had few other things configured in my workspace. With everything cleaned out and using the plugin configured with the following configuration, I have similar results to the one attached:

```
<configuration>
    <configLocation>checkstyle.xml</configLocation>
    <encoding>UTF-8</encoding>
    <consoleOutput>true</consoleOutput>
    <failsOnError>true</failsOnError>
</configuration>
```

The question is should these warning fail the build or should they just warn the developers?

**Comment by Oliver-Loeffler** [comment]
For the beginning we should only issue the warnings.
Then, I would cleanup the existing warnings we have now and as soon we have 0 warnings with the current master we could fail the build.

I like the idea to have the `consoleOutput=true`.

**Comment by Oliver-Loeffler** [comment]
For the tidy up, we should align on small steps, so that we won't get any huge merge conflicts. Possibly this could be achieved by submitting multiple tidyup PRs.

File: README.md [hunk 1]
```diff
@@ -84,3 +84,24 @@ You can add it as a regular dependency to the build of your app:
...
+
+## Code Style
+
+To ensure that new code formatting matches the requirements for Pull Requests,
+the Maven Checkstyle plugin can be used to create a report listing possibly coding 
+style violations.
+
+The [Checkstyle](https://checkstyle.org) configuration is currently in a very early stage and so at present
+it checks for empty blocks, extra white space, padding and empty lines. 
+
```
Comment (abhinayagarwal) [inline README.md:96]: I would reword the above paragraphs to the following:

> Contributors can check for code-style violations in their code by running the [Checkstyle](https://checkstyle.org) Maven goal. The checkstyle configuration is currently in a very early stage and only checks for empty blocks, extra white space, padding and empty lines.

**Comment by abhinayagarwal** [comment]
The maven command for both [Linux](https://github.com/gluonhq/scenebuilder/blob/master/.github/workflows/build.yml#L39) and [Windows/Mac](https://github.com/gluonhq/scenebuilder/blob/master/.github/workflows/build.yml#L45) needs to be updated to run the new checks.

File: README.md (hunk 1 above)
Comment (Oliver-Loeffler) [reply README.md:95]: Yes, this sounds much friendlier, so this scentence is updated accordingly.

File: checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        </module>
+        <module name="RightCurly">
+            <property name="tokens" value="LITERAL_DO"/>
+            <property name="option" value="same"/>
+        </module>
+        <module name="RightCurly">
+            <property name="tokens" value="INTERFACE_DEF"/>
+            <property name="option" value="alone_or_singleline"/>
+        </module>
+        -->
```
Comment (abhinayagarwal) [inline checkstyle.xml:118]: Can we remove this commented block or does it still hold any significance?

Comment (Oliver-Loeffler) [reply checkstyle.xml]: This is not needed so I removed it.
